contracts/services/splitter/src/msg.rs

```diff
@@ -0,0 +1,335 @@
+    }
+}
+
+impl ServiceConfigInterface<ServiceConfig> for ServiceConfig {
+    /// This function is used to see if 2 configs are different
+    fn is_diff(&self, other: &ServiceConfig) -> bool {
+        !self.eq(other)
+    }
+}
+
+impl OptionalServiceConfig {
+    pub fn update_config(self, deps: &DepsMut, config: &mut Config) -> Result<(), ServiceError> {
+        // First update input_addr (if needed)
+        if let Some(input_addr) = self.input_addr {
+            config.input_addr = deps.api.addr_validate(&input_addr)?;
+        }
+
+        // Then validate & update splits (if needed)
+        if let Some(splits) = self.splits {
+            validate_splits(deps.api, &splits)?;
+
+            config.splits = convert_to_checked_configs(deps.as_ref(), &splits)?;
+        }
+        Ok(())
+    }
+}
+
+fn validate_splits(
+    api: &dyn cosmwasm_std::Api,
+    splits: &Vec<UncheckedSplitConfig>,
+) -> Result<(), ServiceError> {
+    if splits.is_empty() {
+        return Err(ServiceError::ConfigurationError(
+            "No split configuration provided.".to_string(),
+        ));
+    }
+
+    let mut denom_map: HashMap<String, ()> = HashMap::new();
+    let mut denom_ratios: HashMap<String, Decimal> = HashMap::new();
+    for split in splits {
+        api.addr_validate(&split.account)?;
+        // Note: can't validate denom without the deps
+
+        // Ensure splits are unique in split configs
+        let key = format!("{:?}|{}", split.denom, split.account);
+        if denom_map.contains_key(&key) {
+            return Err(ServiceError::ConfigurationError(format!(
+                "Duplicate split '{:?}|{}' in split config.",
+                split.denom, split.account
+            )));
+        }
+        denom_map.insert(key.clone(), ());
+
+        match (split.amount, &split.ratio) {
+            (Some(_), None) => {
+                // No need to validate amount
+            }
+            (None, Some(UncheckedRatioConfig::FixedRatio(ratio))) => {
+                denom_ratios
+                    .entry(format!("{:?}", split.denom))
+                    .and_modify(|sum| *sum += ratio)
+                    .or_insert(*ratio);
+            }
+            (None, Some(UncheckedRatioConfig::DynamicRatio { contract_addr, .. })) => {
+                api.addr_validate(contract_addr)?;
+            }
+            (Some(_), Some(_)) | (None, None) => {
+                return Err(ServiceError::ConfigurationError(
+                    "Invalid split config: should specify either an amount or a ratio.".to_string(),
+                ));
+            }
+        }
+    }
+
+    // Verify sum per denom is equal to 1 (rounded up)
+    for (key, sum) in denom_ratios.iter() {
+        if sum.to_uint_ceil() != Uint128::one() {
+            return Err(ServiceError::ConfigurationError(format!(
+                "Invalid split config: sum of ratios for denom '{}' is not equal to 1.",
+                key
+            )));
+        }
+    }
```

Comment:
Wouldnt it be possible that we are using for one output account a ratio and for another output account an amount (no ratio)? I guess that would make the config invalid too right?

What I'm saying is I think we should validate that for a denom we only send ratios or we only send amounts to make sure that we always enough to send. LMK if I misunderstood something